src/main/java/com/arpnetworking/metrics/common/sources/ActorSource.java

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright 2016 Smartsheet
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.metrics.common.sources;
+
+import akka.actor.ActorRef;
+import akka.actor.ActorSystem;
+import akka.actor.Props;
+import com.arpnetworking.metrics.mad.actors.SourceSupervisor;
+import com.fasterxml.jackson.annotation.JacksonInject;
+import net.sf.oval.constraint.NotEmpty;
+import net.sf.oval.constraint.NotNull;
+
+import java.util.function.Function;
+
+/**
+ * Serves as a base calss for actor-based sources.  Actor lifecycle is
+ * handled by the {@link SourceSupervisor}.
+ *
+ * @author Brandon Arp (brandon dot arp at smartsheet dot com)
+ */
+public abstract class ActorSource extends BaseSource {
+    public String getActorName() {
+        return _actorName;
+    }
+
+    public ActorSystem getActorSystem() {
+        return _actorSystem;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void start() {
+        _actorSystem.actorSelection("/user/source").tell(
+                new SourceSupervisor.StartSource(createProps(), _actorName),
+                ActorRef.noSender());
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void stop() {
+        _actorSystem.actorSelection("/user/source").tell(
+                new SourceSupervisor.StopSource(_actorName),
```

Comment:
I am assuming that the supervisor will reconfigure and resurrect this actor should its configuration change. Is there not a race condition here in stop between the supervisor restarting the actor and receiving the stop message?

Haven't gotten to where the actor is stopped, but would it not be easier to stop it by stopping the supervisor? The same is probably true for creation.